src/screens/MainApp/MyDiagnoses/screens/SolutionRequest/index.js

```diff
@@ -27,22 +27,22 @@ const SolutionRequest = ({ navigation }) => {
   const handleUpload = async () => {
     setInputEnabled(false)
     setUploading(true)
+    let doNavigate = false
     try {
       await DatabaseService.addDiagnoseResponse(diagnose.id, description)
       await DatabaseService.setDiagnoseSolvedMark(diagnose.id, true)
 
       setDescription('')
-      setUploading(false)
-      setInputEnabled(true)
-
-      navigation.navigate('FinishScreen')
+      doNavigate = true
```

Comment:
If you have to navigate away of this screen, does it matter if the navigation happens _before_ or _after_ the states are set? I think it does not. IMO the navigation _can_ be here, before setting the states. Also using boolean variables to determining if a `try/catch` succeeded is a code smell (if I remember correctly, I read that on Clean Code).